frontend/app/services/work-package-service.js

```diff
@@ -75,6 +76,47 @@ module.exports = function($http,
   }
 
   var WorkPackageService = {
+    initializeWorkPackage: function(projectIdentifier, initialData) {
+      var changes = _.clone(initialData);
+      if (initialData.type) {
+        changes._links = changes.links || {};
+        changes._links.type = {
+          href: initialData.type
+        };
+        delete changes.type;
+      }
```

Comment:
I don't like the mixed reference to `initialData` and `changes` in this if-block. You should probably always refer to `changes`, since this is the copy you just created for use in this method.